### PR TITLE
refactor: tweak button style code

### PR DESCRIPTION
**Pull request by afc163**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [ ] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [x] 🛠 Refactoring
- [x] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

微调一下，理论无实际影响。

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  --         |
| 🇨🇳 Chinese |  --         |

### CODE REVIEW

**Review by zombieJ — DISMISSED** [review]

File: components/button/style/compact.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import type { GenerateStyle } from '../../theme/internal';
+import { genSubStyleComponent } from '../../theme/internal';
+import type { ButtonToken } from './token';
+import { prepareComponentToken, prepareToken } from './token';
+
+const genButtonCompactStyle: GenerateStyle<ButtonToken> = (token) => {
+  const { componentCls, colorPrimaryHover, lineWidth, calc } = token;
+  const getCompactBorderStyle = (vertical?: boolean) =>
+    ({
+      [`${componentCls}-compact${vertical ? '-vertical' : ''}-item${componentCls}-primary:not([disabled])`]:
```
Comment (li-jia-nan) [inline components/button/style/compact.ts:15]: 感觉分开写可读性好一点：
```ts
const genButtonCompactStyle: GenerateStyle<ButtonToken> = token => {
  const { componentCls, colorPrimaryHover, lineWidth, calc } = token;
  const generalCompactStyle = {
    [`${componentCls}-compact-item, ${componentCls}-compact-vertical-item`]: {
      [`&.${componentCls}-primary:not([disabled])`]: {
        "& + &::before": {
          position: "absolute",
          top: calc(lineWidth).mul(-1).equal(),
          backgroundColor: colorPrimaryHover,
          content: '""',
        },
      },
    },
  };
  const verticalCompactStyle = {
    [`${componentCls}-compact-vertical-item${componentCls}-primary:not([disabled])`]: {
      [`&.${componentCls}-primary:not([disabled])`]: {
        "& + &::before": {
          height: lineWidth,
        },
      },
    },
  };
  const horizontalCompactStyle = {
    [`${componentCls}-compact-item${componentCls}-primary:not([disabled])`]: {
      [`&.${componentCls}-primary:not([disabled])`]: {
        "& + &::before": {
          width: lineWidth,
        },
      },
    },
  };
  return { ...generalCompactStyle, ...verticalCompactStyle, ...horizontalCompactStyle };
};
```

Comment (afc163) [reply components/button/style/compact.ts:15]: 能省一点字节是一点。

File: components/button/style/compact.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        {
+          '& + &::before': {
+            position: 'absolute',
+            top: calc(lineWidth).mul(-1).equal(),
```
Comment (afc163) [inline components/button/style/compact.ts:19]:
```suggestion
            insetInlineStart: calc(lineWidth).mul(-1).equal(),
```

File: components/button/style/compact.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+const genButtonCompactStyle: GenerateStyle<ButtonToken> = (token) => {
+  const { componentCls, colorPrimaryHover, lineWidth, calc } = token;
+  const getCompactBorderStyle = (vertical?: boolean) =>
+    ({
+      [`${componentCls}-compact${vertical ? '-vertical' : ''}-item${componentCls}-primary:not([disabled])`]:
+        {
+          '& + &::before': {
+            position: 'absolute',
+            insetInlineStart: calc(lineWidth).mul(-1).equal(),
```
Comment (afc163) [inline components/button/style/compact.ts:19]:
```suggestion
            top: calc(lineWidth).mul(-1).equal(),
            insetInlineStart: calc(lineWidth).mul(-1).equal(),
```

File: components/button/style/compact.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+const genButtonCompactStyle: GenerateStyle<ButtonToken> = (token) => {
+  const { componentCls, colorPrimaryHover, lineWidth, calc } = token;
+  const getCompactBorderStyle = (vertical?: boolean) =>
+    ({
+      [`${componentCls}-compact${vertical ? '-vertical' : ''}-item${componentCls}-primary:not([disabled])`]:
+        {
+          '& + &::before': {
+            position: 'absolute',
+            top: calc(lineWidth).mul(-1).equal(),
+            insetInlineStart: calc(lineWidth).mul(-1).equal(),
```
Comment (afc163) [inline components/button/style/compact.ts:20]:
```suggestion
            insetInlineStart: 0,
```

File: components/button/style/compact.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+const genButtonCompactStyle: GenerateStyle<ButtonToken> = (token) => {
+  const { componentCls, colorPrimaryHover, lineWidth, calc } = token;
+  const getCompactBorderStyle = (vertical?: boolean) =>
+    ({
+      [`${componentCls}-compact${vertical ? '-vertical' : ''}-item${componentCls}-primary:not([disabled])`]:
+        {
+          '& + &::before': {
+            position: 'absolute',
+            top: calc(lineWidth).mul(-1).equal(),
+            inline: 0,
```
Comment (afc163) [inline components/button/style/compact.ts:20]:
```suggestion
            insetInline: 0,
```

File: components/button/style/compact.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    ({
+      [`${componentCls}-compact${vertical ? '-vertical' : ''}-item${componentCls}-primary:not([disabled])`]:
+        {
+          '& + &::before': {
+            position: 'absolute',
+            top: calc(lineWidth).mul(-1).equal(),
+            [vertical ? 'insetInline' : 'insetBlock']: 0,
+            backgroundColor: colorPrimaryHover,
+            content: '""',
+            [vertical ? 'height' : 'width']: lineWidth,
```
Comment (afc163) [inline components/button/style/compact.ts:23]:
```suggestion
            [vertical ? 'width' : 'height']: '100%',
```

File: components/button/style/compact.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+const genButtonCompactStyle: GenerateStyle<ButtonToken> = (token) => {
+  const { componentCls, colorPrimaryHover, lineWidth, calc } = token;
+  const getCompactBorderStyle = (vertical?: boolean) =>
+    ({
+      [`${componentCls}-compact${vertical ? '-vertical' : ''}-item${componentCls}-primary:not([disabled])`]:
+        {
+          '& + &::before': {
+            position: 'absolute',
+            top: calc(lineWidth).mul(-1).equal(),
+            [vertical ? 'insetInline' : 'insetBlock']: 0,
```
Comment (afc163) [inline components/button/style/compact.ts:20]:
```suggestion
            insetInlineStart: 0,
```

File: components/button/style/compact.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    ({
+      [`${componentCls}-compact${vertical ? '-vertical' : ''}-item${componentCls}-primary:not([disabled])`]:
+        {
+          '& + &::before': {
+            position: 'absolute',
+            top: calc(lineWidth).mul(-1).equal(),
+            insetInlineStart: 0,
+            backgroundColor: colorPrimaryHover,
+            content: '""',
+            [vertical ? 'width' : 'height']: '100%',
```
Comment (afc163) [inline components/button/style/compact.ts:23]:
```suggestion
						width: vertical ? '100%' : lineWidth,
						height: vertical ? lineWidth : '100%',
```

**Comment by afc163** [comment]
帮忙 CR @ant-design/ant-design-collaborators

**Review by li-jia-nan — APPROVED** [review]

**Comment by li-jia-nan** [comment]
diff 看起来好像没区别？

**Comment by afc163** [comment]
两个像素的区别。

**Pull request merged**